screencap.py

```diff
@@ -119,46 +134,46 @@ def retranslateUi(self, mainWindow):
         self.actionmvk.setText(_translate("mainWindow", "mvk"))
 
     def clicked_format(self, extension):
-        f = open('memory_screenshot.txt','r')
-        lst = f.read().split(',')
-        f.close()
-        f = open('memory_screenshot.txt','w')
-        lst[1] = extension
-        f.write('{0},{1},{2}'.format(lst[0],lst[1],lst[2]))
-        f.close()
+        previous = data['image']['extension']
+        data['image']['extension'] = extension
+        if previous != data['image']['extension']:
+            with open('memory.json', 'w') as f:
+                f.seek(0)
```

Comment:
There is no need for `seek`, you are opening a file for writing so you'll always be at the very beginning.